.vscode/tasks.json

```diff
@@ -38,74 +38,81 @@
-            ]
+            "problemMatcher": []
         },
         {
-            "label": "run unit tests",
+            "label": "Kill Speculos",
             "type": "shell",
-            "command": "cd unit-tests && rm -rf  build && cmake -Bbuild -H. && make -C build && CTEST_OUTPUT_ON_FAILURE=1 make -C build test",
+            "command": "pkill -f speculos",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "problemMatcher": [
-                "$gcc"
-            ]
+            "problemMatcher": []
         },
         {
-            "label": "run Speculos",
+            "label": "Run functional tests",
             "type": "shell",
-            "command": "python /path/to/speculos/speculos.py ${workspaceFolder}/bin/app.elf --ontop --sdk 1.6 --apdu-port 9999 --button-port 42000 --automation-port 43000",
+            "command": "pytest tests/ --tb=short -v --device ${input:model}",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "dependsOn": [
-                "make debug"
-            ],
             "problemMatcher": []
         },
         {
-            "label": "[debug] run Speculos",
+            "label": "Install tests requirements",
             "type": "shell",
-            "command": "python /path/to/speculos/speculos.py -d ${workspaceFolder}/bin/app.elf --ontop --sdk 1.6 --apdu-port 9999 --button-port 42000 --automation-port 43000",
+            "command": "pip install -r tests/requirements.txt",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "dependsOn": [
-                "make debug"
-            ],
             "problemMatcher": []
         },
         {
-            "label": "run tests",
+            //  This won't work in Docker for Mac (no support for USB passthrough)
+            "label": "Run make load",
             "type": "shell",
-            "command": "cd tests && pytest",
+            "command": "BOLOS_SDK=${input:sdk} make load",
             "group": {
                 "kind": "build",
                 "isDefault": true
             },
-            "problemMatcher": []
+            "problemMatcher": [
+                "$gcc"
+            ]
         },
+    ],
+    "inputs": [
         {
-            "label": "kill Speculos",
-            "type": "shell",
-            "command": "pkill -f speculos.py",
-            "group": {
-                "kind": "build",
-                "isDefault": true
-            },
-            "problemMatcher": []
-        }
+            "id" : "sdk",
+            "type": "pickString",
+            "description": "Which device do you want to build for ?",
+            "options": [
+                "$NANOS_SDK TARGET=nanos",
+                "$NANOX_SDK TARGET=nanox",
+                "$NANOSP_SDK TARGET=nanos2",
+                "$STAX_SDK TARGET=stax",
```

Comment:
```suggestion
                "$NANOS_SDK",
                "$NANOX_SDK",
                "$NANOSP_SDK",
                "$STAX_SDK",
```
this is enough no?

Reply:
It should be enough indeed as the target is now set by the makefiles of the SDK